src/main/scala/org/constellation/primitives/TransactionGenerator.scala

```diff
@@ -154,7 +157,15 @@ class TransactionGenerator[F[_]: Concurrent: Logger](
     transactionService.put(TransactionCacheData(tx, path = Set(dao.id)))
 
   private def numberOfTransaction(peers: Seq[(Id, PeerData)]): F[Int] =
-    Sync[F].pure { (dao.processingConfig.randomTXPerRoundPerPeer / peers.size + 1) + 1 }
+    roundCounter.getAndIncrement() match {
+      case x if x < transactionsRounds                 => Sync[F].pure(dao.processingConfig.maxTransactionsPerRound)
+      case y if y < (transactionsRounds + emptyRounds) => Sync[F].pure(0)
```

Comment:
IMO it's better to use implicit convention: `0.pure[F]` or `dao.processingConfig.maxTransactionsPerRound.pure[F]`